torch/_dynamo/variables/functions.py

```diff
@@ -932,7 +932,16 @@ def call_function(
         args: "list[VariableTracker]",
         kwargs: "dict[str, VariableTracker]",
     ) -> "VariableTracker":
-        assert is_generator(self.vt.get_code())
+        if not is_generator(self.vt.get_code()):
+            unimplemented_v2(
+                "contextmanager_non_generator",
+                "Function decorated with @contextlib.contextmanager",
+                "Cannot compile generator function decorated with @contextlib.contextmanager that does not use yield",
+                [
+                    "Use 'yield' in the function body instead of 'return'",
+                    "Remove the @contextlib.contextmanager decorator",
+                ],
```

Comment:
```suggestion
                gb_type="non-generator contextlib.contextmanager",
                context=str(self.vt.get_code()),
                explanation="Cannot compile function decorated with `@contextlib.contextmanager` that is not a generator"
                ", i.e. does not use `yield`",
                hints=[
                    "Use `yield` in the function body instead of `return`.",
                    "Remove the `@contextlib.contextmanager` decorator.",
                ],
```